--- a/drivers/net/wireless/rtl8822c/include/osdep_service.h	2020-07-06 14:24:04.424907892 +0300
+++ b/drivers/net/wireless/rtl8822c/include/osdep_service.h	2020-07-06 15:17:02.251397563 +0300
@@ -48,7 +48,7 @@
 	#include <linux/version.h>
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(4, 11, 0))
 	#include <linux/sched/signal.h>
-	#include <linux/sched/types.h>
+	#include <uapi/linux/sched/types.h>
 #endif
 	#include <osdep_service_linux.h>
 	#include <drv_types_linux.h>
