diff --git a/arch/arm64/boot/dts/amlogic/jethome-j80-v1.dts b/arch/arm64/boot/dts/amlogic/jethome-j80-v1.dts
index 2602940c2..86c8f84ec 100644
--- a/arch/arm64/boot/dts/amlogic/jethome-j80-v1.dts
+++ b/arch/arm64/boot/dts/amlogic/jethome-j80-v1.dts
@@ -19,4 +19,33 @@ memory@0 {
 		device_type = "memory";
 		reg = <0x0 0x0 0x0 0x40000000>;
 	};
+
+	// It is necessary only as key map to read efuse values from nvmem from Armbian OS.
+	efusekey:efusekey{
+		keynum = <4>;
+		key0 = <&key_0>;
+		key1 = <&key_1>;
+		key2 = <&key_2>;
+		key3 = <&key_3>;
+		key_0:key_0{
+			keyname = "mac";
+			offset = <0>;
+			size = <6>;
+		};
+		key_1:key_1{
+			keyname = "mac_bt";
+			offset = <6>;
+			size = <6>;
+		};
+		key_2:key_2{
+			keyname = "mac_wifi";
+			offset = <12>;
+			size = <6>;
+		};
+		key_3:key_3{
+			keyname = "usid";
+			offset = <18>;
+			size = <32>;
+		};
+	};
 };
diff --git a/arch/arm64/boot/dts/amlogic/jethome-j100-v1.dts b/arch/arm64/boot/dts/amlogic/jethome-j100-v1.dts
index 1ee7439b9..bdd671acc 100644
--- a/arch/arm64/boot/dts/amlogic/jethome-j100-v1.dts
+++ b/arch/arm64/boot/dts/amlogic/jethome-j100-v1.dts
@@ -279,6 +279,35 @@ data:data
 		};
 	};
 
+	// It is necessary only as key map to read efuse values from nvmem from Armbian OS.
+	efusekey:efusekey{
+		keynum = <4>;
+		key0 = <&key_0>;
+		key1 = <&key_1>;
+		key2 = <&key_2>;
+		key3 = <&key_3>;
+		key_0:key_0{
+			keyname = "mac";
+			offset = <0>;
+			size = <6>;
+		};
+		key_1:key_1{
+			keyname = "mac_bt";
+			offset = <6>;
+			size = <6>;
+		};
+		key_2:key_2{
+			keyname = "mac_wifi";
+			offset = <12>;
+			size = <6>;
+		};
+		key_3:key_3{
+			keyname = "usid";
+			offset = <18>;
+			size = <16>;
+		};
+	};
+
 };
 
 &ethmac {
